kaniko.go

```diff
@@ -174,6 +176,50 @@ func (p Plugin) Exec() error {
 		return fmt.Errorf("repository name to publish image must be specified")
 	}
 
+	if p.Build.LocalTarPath != "" {
+		if p.Build.Repo == "" {
+			return fmt.Errorf("destination repository name must be specified")
+		}
+
+		// Validate local tarball exists
+		if _, err := os.Stat(p.Build.LocalTarPath); os.IsNotExist(err) {
+			return fmt.Errorf("local tarball does not exist at path: %s", p.Build.LocalTarPath)
+		}
+
+		var tags []string
+		var err error
+		if p.Build.AutoTag {
+			tags, err = p.Build.AutoTags()
+			if err != nil {
+				return err
+			}
+		} else if len(p.Build.Tags) > 0 {
+			tags = p.Build.Tags
+		} else {
+			tags = []string{"latest"}
+		}
+
+		// Load the image from the tarball
+		img, err := crane.Load(p.Build.LocalTarPath)
```

Comment:
Are there any size limits for the image? Please test with large images.

Reply:
Yep doing that only right now.

Reply:
I tried with huge pytorch image (~4.5gb just for linux amd64 image) just to check and it pushed successfully to the registry.
execution - https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/all/orgs/default/projects/anshikaci/pipelines/CI15370_CheckingLocalTarballPushKaniko/deployments/bd-HHwVRQf6rMR8Gbt5qFQ/pipeline?storeType=INLINE&step=4pm27Kp8SFer4TxPrI8YhA&stage=zmENeIDmQa-c_B5fGwcrXA&childStage=&stageExecId=